via/services/checkmate.py

```diff
@@ -24,20 +31,23 @@ def raise_if_blocked(self, url):
         :raises BadURL: For malformed or private URLs
         """
         try:
-            blocked = self.check_url(
-                url,
-                allow_all=self._request.registry.settings["checkmate_allow_all"],
-                blocked_for=self._request.params.get("via.blocked_for"),
-                ignore_reasons=self._request.registry.settings[
-                    "checkmate_ignore_reasons"
-                ],
-            )
-
+            blocked = self.check_url(url)
         except CheckmateBadURL as exc:
             raise BadURL(exc, url=url) from exc
-
         except CheckmateException:
             blocked = None
 
         if blocked:
             raise HTTPTemporaryRedirect(location=blocked.presentation_url)
+
+
+def factory(_context, request):
```

Comment:
```suggestion
def service_factory(_context, request):
```

A quick future compatibility tip. I don't think it'll ever get there, but I think we are standardizing on `service_factory` as the name to allow for splitting big services into multiple files.

It's a bit of a nightmare to import `blah.factory` if it's both a method and a module name.

Reply:
I'm gonna leave this because they're actually _all_ called `factory()` not `service_factory()` in Via currently, so perhaps there needs to be a separate PR to rename them all consistently. I suspect splitting things into multiple _services_ would be a better pattern, rather than splitting something into a single big service with multiple files.